modules/cluster-cloud-controller-config-osp.adoc

```diff
@@ -0,0 +1,271 @@
+
+// | `cascade-delete`
+// | Determines whether or not to perform cascade deletion of load balancers. Default: true.
+
+// | `flavor-id`
+// | The id of the loadbalancer flavor to use. Uses octavia default if not set.
+
+// | `availability-zone`
+// | The name of the loadbalancer availability zone to use. It is applicable if use-octavia is set to True and requires Octavia API version 2.14 or later (Ussuri release). The Octavia availability zone capabilities will not be used if it is not set. The parameter will be ignored if the Octavia version doesn't support availability zones yet.
+
+| `LoadBalancerClass "ClassName"`
+| This is a config section that comprises a set of options:
+
+ * `floating-network-id`
+ * `floating-subnet-id`
+ * `floating-subnet`
+ * `floating-subnet-tags`
+ * `network-id`
+ * `subnet-id`
+//  * `member-subnet-id`
+
+ These options behavior is identical to what is described in the identically named load balancer options in the load balancer section of the CCM config file.
+
+ You can set the `ClassName` value by specifying the Service annotation `loadbalancer.openstack.org/class`. 
+
+// | `enable-ingress-hostname`
+// | Used with proxy protocol (set by annotation `loadbalancer.openstack.org/proxy-protocol: "true"`) by adding a dns suffix (nip.io) to the load balancer IP address. Default false.
+
+ This option is currently a workaround for the issue https://github.com/kubernetes/ingress-nginx/issues/3996, should be removed or refactored after the Kubernetes [KEP-1860](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1860-kube-proxy-IP-node-binding) is implemented.
+
+// | `ingress-hostname-suffix`
+// | The dns suffix to the load balancer IP address when using proxy protocol. Default nip.io
+
+ This option is currently a workaround for the issue https://github.com/kubernetes/ingress-nginx/issues/3996, should be removed or refactored after the Kubernetes [KEP-1860](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1860-kube-proxy-IP-node-binding) is implemented.
+
+// | `default-tls-container-ref`
+// | Reference to a tls container. This option works with Octavia, when this option is set then the cloud provider will create an Octavia Listener of type TERMINATED_HTTPS for a TLS Terminated loadbalancer.
+
+//  Format for tls container ref: `https://{keymanager_host}/v1/containers/{uuid}`
+//  Check `container-store` parameter if you want to disable validation.
+
+// | `container-store`
+// | Optional. Used to specify the store of the tls-container-ref, e.g. "barbican" or "external" - other store will cause a warning log. Default value - `barbican` - existence of tls container ref would always be performed. If set to `external` format for tls container ref will not be validated.
+
+| `max-shared-lb`
+| The maximum number of Services that can share a load balancer. The default value is `2`.
+|===
+
+[id="ccm-config-metadata-options"]
+== Metadata options
```

Comment:
I don't think these options are tested

Reply:
True, it was not tested. I think we should remove it from here.